src/MgvOfferMaking.sol

```diff
@@ -304,7 +304,13 @@ contract MgvOfferMaking is MgvHasOffers {
       require(TickLib.inRange(insertionTick), "mgv/writeOffer/tick/outOfRange");
 
       // must cache tick because branch will be modified and tick information will be lost (in case an offer will be removed)
-      Tick cachedLocalTick = ofp.local.bestTick();
+      Tick cachedLocalTick;
+      // force control flow through gas saving path if offer list is empty
```

Comment:
Would be nice with an explanation of how this works.

If I understand correctly, the trick is that `cachedLocalTick == insertionTick  =>  insertionIndex == currentIndex` which means we don't read&write level{0,1} in mappings below.

Reply:
I think this also means that we don't need to check whether `level2` is empty in the code below (GH won't let me comment on the lines in question, it's L369 and L389).

Ie I think we can change
```
if (insertionIndex <= currentIndex || ofp.local.level2().isEmpty()) {
```

into 
```
if (insertionIndex <= currentIndex) {
```

since `ofp.local.level2().isEmpty()  =>  ofp.local.level0().isEmpty()  =>  cachedLocalTick == insertionTick  =>  insertionIndex == currentIndex`.

Reply:
Yes I forgot to remove those! Will do when I’m back

Reply:
Addressed in d077234bb08e93993c783350582c05565c014869 :)